content/articles/2019-05-itdepends.Rmarkdown

```diff
@@ -0,0 +1,241 @@
+---
+title: It depends - A dialog about dependencies
+slug: itdepends
+description: >
+    It depends - A dialog about dependencies
+date: 2019-05-14
+author: Jim Hester
+photo:
+  url: ???
+  author: ???
+categories: [???]
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE, comment = "#>", 
+  fig.width = 7, 
+  fig.align = 'center',
+  fig.asp = 0.618, # 1 / phi
+  out.width = "700px"
+)
+```
+
+# It depends - A dialog about dependencies
+
+> This post paraphrases my [talk with the same name](https://resources.rstudio.com/rstudio-conf-2019/it-depends-a-dialog-about-dependencies) at RStudio::conf 2019.
+
+Software dependencies can often be a double edged sword.
+
+On one hand they let you take advantage of others' work, giving your software
+marvelous new features and reducing bugs.
+
+On the other hand they can change over time, causing your software to break
+unexpectedly and (potentially) increasing your maintenance burden. They can also
+ take additional disk space and installation time.
+
+
+These downsides have led some to suggest a 'dependency zero' mindset.
+
+I feel this is a major mistake for most projects, and will instead increase
+maintenance, lower functionality and introduce additional bugs.
+
+## dependencies are not equal
+
+One issue with striving for minimal dependencies is it treats all dependencies
+as equivalent. However in reality this is far from the case. There are many
+axis upon which dependencies can differ, but some of the most important include
+
+1. The number of additional dependencies. For example the
```

Comment:
When you say "additional" dependencies here, at first I thought you were getting at "marginal addition". You're not -- I see that now.

But the marginal angle is one you might consider too. "If I depend on package X, what actually changes for me and my users?" Various scenarios:

  * Package X is already in my dependencies, implicitly, so making it explicit doesn't really change anything.
    - Depending on a Recommended package that ships with R.
    - Depending on a hard dependency (stringi) of one of your hard dependencies (stringr). Yes, technically, this could change but how likely is that and on what time scale.
  * Package X is, empirically speaking, already installed on 90% of my users' machines, due to patterns of actual package co-usage. So making it an explicit dependency causes very few new installations.

Reply:
Maybe "downstream" dependencies would be a better term here? 

And @jennybc point deserves discussion somewhere.

Reply:
Yes agreed on this being something we should mention